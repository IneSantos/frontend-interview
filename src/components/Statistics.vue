<template>
  <section class="statistics"  id="statistics">
    <div class="header"> 
        <h1>Awesome statistics</h1>
        <p>All statistics in on place!</p>
    </div>
    <div class="percentages">
        <div class="game-visctories">
            <h1>Game victories %</h1>
            <div class="players">
                <div class="stats-container">
                    <p>Player 1</p>
                    <div class="victory-loss">
                        <div class="stats">
                            <div class="statsCircle" :class="[getColor(winsPlayer1)]">{{winsPlayer1}}%</div>
                            <p>V</p>
                        </div> 
                        <div class="stats">
                            <div class="statsCircle" :class="[getColor(lossPlayer1)]">{{lossPlayer1}}%</div>
                            <p>L</p>
                        </div> 
                    </div> 
                </div>  
                <div class="stats-container">
                    <p>Player 2</p>
                    <div class="victory-loss">
                        <div class="stats">
                            <div class="statsCircle" :class="[getColor(winsPlayer2)]">{{winsPlayer2}}%</div>
                            <p>V</p>
                        </div> 
                        <div class="stats">
                            <div class="statsCircle" :class="[getColor(lossPlayer2)]">{{lossPlayer2}}%</div>
                            <p>L</p>
                        </div>
                    </div> 
                </div> 
            </div>  
        </div>
        <div class="stat-section">
            <h1>Played matchs</h1>
            <div class="dots"> 
                <div v-for="(winner, index) in gameHistory" :key="index">
                    <span class="dot" :class="[winner !== '' ? 'active' : '']"></span>
                </div>
            </div>
        </div>
        <div class="stat-section game-history">
            <h1>Game history</h1>
            <div class="history"> 
                <div v-for="(winner, index) in gameHistory" :key="index" class="winner">
                    {{winner}}
                </div>
            </div>
            
        </div>
        <div class="stat-section">
            <h1>Total time</h1>
            <StopWatch ref="stopwatch"/>
        </div>
    </div>
  </section>
</template>


<script>
import '@/styles/statistics.scss';
import StopWatch from './tic-tac-toe//StopWatch';

export default {
  name: 'Statistics',
  components: {
      StopWatch
  },
  computed: {
    winsPlayer1 () {
        const wins = +this.$store.getters.getWinsPlayer1;
        return (wins/5) * 100;
    },
    lossPlayer1 () {
        const loss = +this.$store.getters.getWinsPlayer1;
        return 100 - ((loss/5) * 100);
    },
    winsPlayer2 () {
        const wins = +this.$store.getters.getWinsPlayer2;
        return (wins/5) * 100;
    },
    lossPlayer2 () {
        const loss = +this.$store.getters.getWinsPlayer2;
        return 100 - ((loss/5) * 100);
    },
    hasGameStarted () {
        return this.$store.getters.getHasGameStarted;
    },
    gameHistory () {
        return this.$store.getters.getGameHistory;       
    }
  },
  watch: {
    hasGameStarted: function () {
       this.$refs.stopwatch.start();
    }
  },
    methods: {
        getColor: function(val) {
            if(val === 0) return 'red';
            if(val > 50) return 'green';
            if(val > 0) return 'yellow';
        }
    }
}
</script>