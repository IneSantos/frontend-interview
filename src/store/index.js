export const store = {
    state: {
      hasgameStarted: false,
      numOfgames: 0,
      winsPlayer1: 0,
      winsPlayer2: 0,
      gameHistory: ['', '', '' , '', '']
    },
    mutations: {
      SET_GAME_STARTED(state, hasgameStarted) {
        state.hasgameStarted = hasgameStarted
      },
      INC_NUM_GAMES(state) {
        state.numOfgames++
      },
      SET_WINS_PLAYER_1(state, winsPlayer1) {
        state.winsPlayer1 = winsPlayer1
      },
      SET_WINS_PLAYER_2(state, winsPlayer2) {
        state.winsPlayer2 = winsPlayer2
      },
      SET_GAME_HISTORY(state, player) {
        state.gameHistory[state.numOfgames-1] = player;
      },
      RESET_GAME(state) {
        state.hasgameStarted = false;
      }
    },
    actions: {
      setGameStart(context, hasgameStarted){
        context.commit('SET_GAME_STARTED', hasgameStarted);
      },
      resetBoard(context){
        context.commit('RESET_BOARD')
      },
      incNumOfGames(context){
        context.commit('INC_NUM_GAMES');
      },
      setWinsPlayer1(context, winsPlayer1){
        context.commit('SET_WINS_PLAYER_1', winsPlayer1);
      },
      setWinsPlayer2(context, winsPlayer2){
        context.commit('SET_WINS_PLAYER_2', winsPlayer2);
      },
      setGameHistory(context, player){
        context.commit('SET_GAME_HISTORY', player);
      }
    },
    getters: {
      getHasGameStarted(state) {
        return state.hasgameStarted;
      },
      getNumOfGames(state) {
        return state.numOfgames;
      },
      getWinsPlayer1(state) {
        return state.winsPlayer1;
      },
      getWinsPlayer2(state) {
        return state.winsPlayer2;
      },
      getGameHistory(state) {
        return state.gameHistory;
      },
    }

}

