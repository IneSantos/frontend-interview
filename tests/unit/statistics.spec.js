import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import Statistics from '@/components/Statistics.vue'
import { storeMocked } from '../__mocks__/store.js'

const localVue = createLocalVue();
localVue.use(Vuex);

const store = new Vuex.Store(storeMocked);

describe('Statistics.vue', () => {
  it('renders default', () => {
    const wrapper = shallowMount(Statistics, { store, localVue });
    expect(wrapper).toMatchSnapshot();
  })
})
