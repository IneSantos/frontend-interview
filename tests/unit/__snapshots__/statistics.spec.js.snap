// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Statistics.vue renders default 1`] = `
<section id="statistics" class="statistics">
  <div class="header">
    <h1>Awesome statistics</h1>
    <p>All statistics in on place!</p>
  </div>
  <div class="percentages">
    <div class="game-visctories">
      <h1>Game victories %</h1>
      <div class="players">
        <div class="stats-container">
          <p>Player 1</p>
          <div class="victory-loss">
            <div class="stats">
              <div class="statsCircle red">0%</div>
              <p>V</p>
            </div>
            <div class="stats">
              <div class="statsCircle green">100%</div>
              <p>L</p>
            </div>
          </div>
        </div>
        <div class="stats-container">
          <p>Player 2</p>
          <div class="victory-loss">
            <div class="stats">
              <div class="statsCircle red">0%</div>
              <p>V</p>
            </div>
            <div class="stats">
              <div class="statsCircle green">100%</div>
              <p>L</p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div class="stat-section">
      <h1>Played matchs</h1>
      <div class="dots">
        <div><span class="dot active"></span></div>
        <div><span class="dot active"></span></div>
        <div><span class="dot active"></span></div>
        <div><span class="dot"></span></div>
        <div><span class="dot"></span></div>
      </div>
    </div>
    <div class="stat-section game-history">
      <h1>Game history</h1>
      <div class="history">
        <div class="winner">
          P1
        </div>
        <div class="winner">
          P1
        </div>
        <div class="winner">
          P2
        </div>
        <div class="winner">

        </div>
        <div class="winner">

        </div>
      </div>
    </div>
    <div class="stat-section">
      <h1>Total time</h1>
      <stopwatch-stub></stopwatch-stub>
    </div>
  </div>
</section>
`;
